R/plot-individual-time-profile.R

```diff
@@ -0,0 +1,118 @@
+#' Concentration time profile plot for individual
+#'
+#' @param dataCombined A `DataCombined` object.
+#' @param defaultPlotConfiguration A `DefaultPlotConfiguration` object, which is
+#'   an `R6` class object that defines plot properties.
+#'
+#' @import tlf
+#'
+#' @family plotting
+#'
+#' @examples
+#'
+#' # TODO: add example
+#'
+#' @export
+plotIndividualTimeProfile <- function(dataCombined,
+                                      defaultPlotConfiguration = NULL) {
+  # validation -----------------------------
+
+  defaultPlotConfiguration <- defaultPlotConfiguration %||% DefaultPlotConfiguration$new()
+  validateIsOfType(dataCombined, "DataCombined")
+  validateIsSameLength(objectCount(dataCombined), 1L) # only single instance is allowed
+  validateIsOfType(defaultPlotConfiguration, "DefaultPlotConfiguration", nullAllowed = FALSE)
+
+  # data frames -----------------------------
+
+  df <- dataCombined$toDataFrame()
+
+  # Getting all units on the same scale
+  df <- .unitConverter(df, defaultPlotConfiguration$xUnit, defaultPlotConfiguration$yUnit)
+
+  # Datasets which haven't been assigned to any group will be plotted as a group
+  # on its own. That is, the `group` column entries for them will be their names.
+  df <- .addMissingGroupings(df)
+
+  # `TimeProfilePlotConfiguration` object -----------------------------
+
+  # Create an instance of `TimeProfilePlotConfiguration` class by doing a
+  # one-to-one mapping of internal plot configuration object's public fields
+  populationTimeProfilePlotConfiguration <- .convertGeneralToSpecificPlotConfiguration(
+    data = df,
+    specificPlotConfiguration = tlf::TimeProfilePlotConfiguration$new(),
+    generalPlotConfiguration = defaultPlotConfiguration
+  )
+
+  # plot -----------------------------
+
+  # Which dataset types are present?
+  datasetTypePresent <- .extractPresentDatasetTypes(dataCombined)
+
+  # both observed and simulated
+  if (datasetTypePresent == .presentDataTypes$Both) {
```

Comment:
Why distinguish between the three cases? Does this logic come from TLF?

Reply:
Yes, the exact _tlf_ function call to create a plot depends on whether both simulated and observed data are present, or just one of them.

Reply:
Great suggestion!

I actually never tried setting it to `NULL`, so missed out on this.